zjit/src/codegen.rs

```diff
@@ -2062,13 +2074,8 @@ fn function_stub_hit_body(cb: &mut CodeBlock, iseq_call: &IseqCallRef) -> Result
         debug!("{err:?}: gen_iseq failed: {}", iseq_get_location(iseq_call.iseq.get(), 0));
     })?;
 
-    // We currently don't support JIT-to-JIT calls for ISEQs with optional arguments.
-    // So we only need to use jit_entry_ptrs[0] for now. TODO(Shopify/ruby#817): Support optional arguments.
-    let Some(&jit_entry_ptr) = jit_entry_ptrs.first() else {
-        return Err(CompileError::JitToJitOptional)
-    };
-
     // Update the stub to call the code pointer
+    let jit_entry_ptr = jit_entry_ptrs[iseq_call.insn_idx as usize];
```

Comment:
This probably needs an index conversion. It's supposed to be used with `jit_entry_insns(iseq)`. `jit_entry_insns(iseq)` is a list of `insn_idx`, and the index of `iseq_call.insn_idx` in `jit_entry_insns(iseq)` should be the index to look up `jit_entry_ptrs`.

Reply:
Going to take a minute to wrap my head around this